pyro/infer/mcmc/api.py

```diff
@@ -103,17 +103,17 @@ def run(self, *args, **kwargs):
             self.result_queue.put_nowait((self.chain_id, e))
 
 
-def _gen_samples(kernel, warmup_steps, num_samples, hook, *args, **kwargs):
+def _gen_samples(kernel, warmup_steps, num_samples, hook, chain_id, *args, **kwargs):
     kernel.setup(warmup_steps, *args, **kwargs)
     params = kernel.initial_params
     # yield structure (key, value.shape) of params
     yield {k: v.shape for k, v in params.items()}
     for i in range(warmup_steps):
         params = kernel.sample(params)
-        hook(kernel, params, 'warmup', i)
+        hook(kernel, params, 'Warmup [{}]'.format(chain_id) if chain_id is not None else 'Warmup', i)
```

Comment:
I think it is a good idea to pass in chain information for additional logging, but is it needed for sequential chaining?

Reply:
Oh, I am fine removing this. I do this to make it consistent with MultiSampler progbar. Probably this is just helpful for visualization if num_chains=10 or so. WDYT?

Reply:
That sounds reasonable, thanks for explaining.